File: samples_generated_code/sample_2/phi4/baseline_code_java/code_row_108.java
import java.io.File;
import java.io.FileInputStream;
import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;
import java.nio.file.Paths;

public class SafeFileReader {

    private static final String DIRECTORY_PATH = "/safe/";

    public static void main(String[] args) {
        if (args.length != 1) {
            System.out.println("Usage: java SafeFileReader <filename>");
            return;
        }

        String filename = args[0];
        Path filePath = Paths.get(DIRECTORY_PATH, filename);

        try {
            // Check if file exists and is a regular file
            if (Files.exists(filePath) && Files.isRegularFile(filePath)) {
                readFileContent(filePath);
            } else {
                System.out.println("The requested file does not exist or is not a regular file.");
            }
        } catch (IOException e) {
            System.out.println("An error occurred while reading the file: " + e.getMessage());
        }
    }

    private static void readFileContent(Path filePath) throws IOException {
        try (FileInputStream fis = new FileInputStream(filePath.toFile())) {
            byte[] buffer = new byte[1024];
            int bytesRead;
            
            while ((bytesRead = fis.read(buffer)) != -1) {
                String content = new String(buffer, 0, bytesRead);
                System.out.print(content);
            }
        }
    }
}

File: samples_generated_code/sample_3/phi4/baseline_code_py/code_row_108.py
import os

def read_safe_file(filename):
    safe_directory = '/safe/'
    filepath = os.path.join(safe_directory, filename)
    
    if not os.path.exists(filepath):
        raise FileNotFoundError(f"The file '{filename}' does not exist in {safe_directory}.")
    
    with open(filepath, 'r') as file:
        content = file.read()
        
    return content

# Example usage
try:
    content = read_safe_file('example.txt')
    print(content)
except Exception as e:
    print(e)